File: src/lib-common/find.js
function find(x,y)
{
 check(is_arr,x)
 check(is_def,y)
 
 for(const k in x)
 {
  const i=to_i(k)
  const v=at(x,i)
  
  if(same(v,y))
   return i
 }
 
 return -1
}

File: src/lib-common/find.cs
fn find x:composite y:def
 if is_str x
  check is_str y

 if is_vec x
  ret x.indexOf y
 elseif is_obj x
  forin x
   let v get x k

   if same v y
    ret k
  end

  ret -1
 else
  stop
end